nyudlts/ultraviolet
# Invenio84 file limits msg - vanilla JS custom code

[Jira ticket INVENIO-84](https://jira.nyu.edu/browse/INVENIO-84)

Similar to #189 but duplicates the notification bar via vanilla JS but by including it in the webpack compilation process of the python project through the sites directory.

Based off of the instructions in InvenioRDM documentation:
- https://inveniordm.docs.cern.ch/develop/howtos/custom_code/
- https://inveniordm.docs.cern.ch/develop/howtos/add_javascript/

This PR will add the following:
- deposit form > file upload: additional warning message about file size and file quantity.
- helper function `waitForElement` that can be reused when waiting for a specific react element to be rendered

To test this PR:
- `invenio-cli services start`
- `invenio-cli packages install site` (runs assets build after installing)
- `invenio-cli run`
- manual testing:
  - navigate to https://127.0.0.1/5000/uploads/new
  - deposit form > under Files > after the Drag and drop zone: should display 2 different message bars:
    - "File addition, removal or modification are not allowed after you have published your upload."
    - "Contact the NYU Libraries UltraViolet service team at [uv@nyu.edu](mailto:email@example.com) if you have more than 100 files or 50GB."
- automated testing:
  - https://github.com/dismorfo/ultraviolet-nightwatch/pull/2

Thank you for your review
